doc/Language/regexes.pod6

```diff
@@ -1643,62 +1650,71 @@ to emphasize the correspondence between the first two statements. Statement
+table as in statement C<[8]>, in which case the match pattern C<string> first
+emerges as the return value of the C<flip> method call. Since general Perl6
+code may be run from within the parentheses of C<$( )>, the same effect can
+also be achieved with a bit more effort, like in statement C<[9]>. Statement
+C<[10]> illustrates how the stringified version of the code's return value (the
+boolean value C<False>) is matched literally.
+
+Finally, statements C<[11]> and C<[12]> show how the value of C<$pattern4> and
+the return value of C<f1> are I<not> subject to a further round of
+interpolation. Hence, in general, after possible stringification, C«$variable»
+and C«$(code)» provide for a strictly literal match of the variable or return
+value.
 
 Now consider the second two syntactical forms from the table above:
 C«<$variable>» and C«<${code}>». These forms will stringify the value of the
 variable or the return value of the code and interpolate it as a regex. If the
-respective value is a Regex, it is interpolated as such. 'Interpolated as a
-regex' means interpolated/inserted into the target Regex without protective
-quoting. Consequently, the further evaluation of the target Regex may trigger
-the (second-level) interpolation of any variables it contains.   
+respective value is a Regex, it is interpolated as such:
 
     my $string         = 'Is this a regex or a string: 123\w+$x ?';
     my $pattern1       = '\w+';
     my $number         = 123;
     my sub f1          { return /s\w+/ };
-    my sub f2 (Str $x) { return /$x x/ };
-    my sub f3          { return Q[$x]  };
 
     say $string.match: / <$pattern1>  /;                  # [1] OUTPUT: ｢Is｣
     say $string.match: / <$number>    /;                  # [2] OUTPUT: ｢123｣
     say $string.match: / <{ f1 }>     /;                  # [3] OUTPUT: ｢string｣
 
-    my $x = "rege";
-    say $string.match: / <{ f2($x) }> /;                  # [4] OUTPUT: ｢regex｣
-    say $string.match: / <{ f3     }> /;                  # [5] OUTPUT: ｢rege｣
-
-In statement C<[4]> use is made of the function C<f2>, which acts as a (very
-simple) "regex factory": you can pass it a string variable, and it will return a
-Regex object into which the variable has been interpolated. In this case, C<f2>
-appends the letter 'x' to whatever string it is passed. The Regex that is
-returned by C<f2> is in turn inserted into target Regex by the C<{...}>
-construct. Statement C<[5]> illustrates another case of two-fold regex
-interpolation. When the target Regex is constructed, the strictly literal string
-value C<$x> is interpolated into it by the C<{...}> construct. When the Regex is
-evaluated further, the unprotected variable C<$x> is interpolated, i.e. replaced
-by the string value C<rege>, which explains the match.
+Importantly, 'interpolated as a regex' means interpolated/inserted into the
+target Regex without protective quoting. Consequently, if the value of the
```

Comment:
Use simply regex or C<Regex>, depending on whether you're talking about the concept or the object.